<?xml version="1.0"?>
<launch>
    <arg name="map_frame_id" default="map"/>
    <arg name="base_link_frame_id" default="base_link"/>
    <arg name="loop_rate" default="50"/>
    <arg name="filePath" value="$(find rumba_autocar)/csv/way1"/>

    <include file="$(find rumba_autocar)/launch/basicNavigation.launch">
        <arg name="nav_mode" default="mapping"/>
        <arg name="move_base" default="false"/>
    </include>

    <!--joystick-->
    <node name="joystickPS4_rumba" pkg="rumba_autocar" type="joystickPS4_rumba">
        <remap from="cmd_vel" to="roomba/cmd_vel"/>
    </node>
    <node name="joy_node" pkg="joy" type="joy_node"/>

    <node name="wpRecord" pkg="rumba_autocar" type="wpRecord" output="screen">
        <param name="map_frame_id" value="$(arg map_frame_id)"/>
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)"/>
        <param name="loop_rate" value="$(arg loop_rate)"/>
        <param name="wayPoint_pitch" value="0.15"/>
    </node>

    <node name="csv_out" pkg="rumba_autocar" type="csv_out">
        <param name="filePath" value="$(arg filePath)"/>
        <remap from="csv_array"  to="path2csv"/>
    </node>

    <!--rviz-->
    <arg name="rvizconfig" default="$(find rumba_autocar)/rviz/rumba_autocar.rviz" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
    
</launch>